File: backend/src/main/java/iuh/labbooking/service/password/PasswordService.java
package iuh.labbooking.service.password;

import iuh.labbooking.dto.request.password.ChangePasswordRequest;
import iuh.labbooking.dto.request.password.ForgotPasswordRequest;
import iuh.labbooking.dto.request.password.ResetPasswordRequest;
import iuh.labbooking.dto.response.otp.SendOtpResponse;

public interface PasswordService {

    SendOtpResponse forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    void changePassword(ChangePasswordRequest request);
}

File: frontend/src/services/password.service.ts
import http from '@/lib/http'
import type { ChangePasswordRequest, ForgotPasswordRequest, ResetPasswordRequest } from '@/schemas/password.schema'
import type { SendOtpResponse } from '@/schemas/otp.schema'
import type { ApiResponse } from '@/schemas/base.schema'

const passwordService = {
  change: (body: ChangePasswordRequest) => http.post<ApiResponse<void>>('/password/change', body),

  forgot: (body: ForgotPasswordRequest) => http.post<ApiResponse<SendOtpResponse>>('/password/forgot', body),

  reset: (body: ResetPasswordRequest) => http.post<ApiResponse<void>>('/password/reset', body)
}

export default passwordService
